ui/src/geoviite-design-lib/dialog/dialog.tsx

```diff
@@ -15,6 +15,10 @@ export enum DialogWidth {
     THREE_COLUMNS = 'dialog__popup--three-columns',
 }
 
+export enum DialogHeight {
+    RESTRICTED_TO_HALF_OF_VIEWPORT = 'dialog__popup--height-restricted-to-half',
```

Comment:
Oliko tästä puhetta että halutaanko tämä tehdä juurikin cappaamalla itse dialogin korkeus? 🤔 Tai siis vähintäänkin voisi varmaan olla hyvä että dialogi ei koita olla korkeampi kuin viewport, mutta toinen vaihtoehtohan juuri tähän spesifiin dialogiin olisi se, että tehdään vain listasta scrollautuva ja määritellään sille jonkinlainen maksimikorkeus. Scrollautuvat listat on myös tapa, jota me ollaan suosittu muissakin Geoviitteen näkymissä, joten sen puolesta suosisin sitä ehkä täälläkin (nythän myös selitteet scrollautuvat kaiken muun mukana)

Reply:
Näytin tätä dailyssä, todettiin tässä vaiheessa UX:n puolesta ihan riittävän järkeväksi näin👍. Hifistelyominaisuutena jatkossa se kysymys voisi olla ns. "subheaderina" fiksattuna aina näkyviin vaikka dialogia scrollaakin (tämä taisi olla siis se mitä hait?). 

Dialogillahan on siis headeri (aina näkyvissä), content-boksi, jossa kysymys myös sijaitsee (scrollattava) sekä footer napeille. Overflowhan (eli scrollaus) onkin sillä content-boksilla, jossa lista sijaitsee, joten käytännössähän se sitä listaa scrollaa jo nytkin (tosin kysymys menee siis piiloon kuten tulkitsitkin aiemmin, muttei sitä koettu liialliseksi haitaksi UX:n puolesta tähän väliin).

Se että onko dialogilla maksimikorkeus vai content-boksilla maksimikorkeus on tosin kysymys itsessään, enpäs tiedä kumpi mahtaisi olla järkevämpi paikka se asettaa 🤔. Tosin nyt kun tässä asetellaan koko dialogille 50% viewportin korkeudesta, niin se ei enää olisi 50% viewportin korkeudesta jos dialog-contentille asetettaisiin se sama 50% max-height. Tiedäppä sitten onko tämä edes erityisen huomattava ero. Periaatteessa nyt voi myös käydä niin, että contentille ei jää pikseliäkään jos viewport on hyvin matala, kun taas jos contentilla olisi 50% viewport max-height, niin dialogi saattaisi alhaisella viewportin korkeudella hypätä yli viewportista (koska header/footer ei skaalaudu korkeuden osalta).

Reply:
Meinasin juu tuota kysymyksen scrollautumista. En ehkä itse lähtisi tuollaiseen subheader-kikkailuun dialogin itsensä tasolla (jollei sitten tehdä jotain erillistä dialogikomponenttia sellaisia tilanteita varten,) kun meillähän on myös dialogeja joissa sisällön koko on about täysin staattinen (muut confirm-dialogit ja edit-dialogit.) Nyt kun kävin itse testaamassa tuota mainin kärjessä, niin ihan hyvä tuo on kyllä noinkin. Lähinnä nostin tuon alunperin esille sillä, että meillä monessa paikassa muualla on haluttu saada pelkästään se juttuja listaava komponentti (lista, taulukko jne.) scrollaamaan samalla kun sen yläpuolella oleva sisältö pysyy paikallaan, niin intuitio sanoi että tässäkin oltaisiin ehkä haluttu vastaavaa. Mutta tämä on tosiaan näin munkin mielestäni OK kun tuota livenä testailin 👍 

En ole ihan varma noista maksimikorkeuksista, pitäisi varmaan testailla pienellä ruudulla että aiheuttaako jompikumpi ongelmia 🤔 Tapauksessa kuin tapauksessa voisi olla kiva että dialogit ylipäätään eivät voi olla korkeampia kuin viewport. Nykyisellään korkeudeltaan uncapatut dialogit eivät taida overflowata mitenkään kohtuuella, vaan piirtyvät niin korkeiksi kuin haluavat (ehkä?) Jos noin, niin noille wanhanmallisille unapatuille dialogeille voisi harkita jonkinlaista maksimikorkeutta jonka jälkeen sisältö alkaa scrollaamaan pakostakin